src/mapping/PartitionOfUnityMapping.hpp

```diff
@@ -0,0 +1,385 @@
+    outMesh    = this->output(); // local
+  }
+
+  if (outMesh->vertices().empty())
+    return; // Ranks not at the interface should never hold interface vertices
+
+  // TODO: Check again the tagging in combination with the partition construction (which mesh to use)
+  // In order to construct the local partitions, we need all vertices with a distance of 2 x radius,
+  // as the relevant partitions centers have a maximum distance of radius, and the proper construction of the
+  // interpolant requires all vertices with a distance of radius from the center.
+  auto bb = outMesh->getBoundingBox();
+
+  if (averagePartitionRadius == 0)
+    averagePartitionRadius = partitioner::estimatePartitionRadius(_verticesPerPartition, filterMesh, bb);
+
+  // @TODO: This assert is not completely right, as it checks all dimensions for non-emptyness (which might not be the case).
+  // However, with the current BB implementation, the expandBy function will just do nothing.
+  PRECICE_ASSERT(!bb.empty());
+  PRECICE_ASSERT(averagePartitionRadius > 0);
+  // Now we extend the bounding box by the radius
+  bb.expandBy(1 * averagePartitionRadius);
+
+  // ... and tag all affected vertices
+  auto verticesNew = filterMesh->index().getVerticesInsideBox(bb);
+
+  std::for_each(verticesNew.begin(), verticesNew.end(), [&filterMesh](VertexID v) { filterMesh->vertices()[v].tag(); });
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void PartitionOfUnityMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshSecondRound()
+{
+  // Probably nothing to be done here. There is no global ownership for matrix entries required and we tag all potentially locally relevant vertices already in the first round.
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void PartitionOfUnityMapping<RADIAL_BASIS_FUNCTION_T>::computeMapping()
+{
+  PRECICE_TRACE();
+
+  precice::utils::Event e("map.pou.computeMapping.From" + this->input()->getName() + "To" + this->output()->getName(), precice::syncMode);
+
+  // Recompute the whole partitioning
+  this->clear();
+
+  mesh::PtrMesh inMesh;
+  mesh::PtrMesh outMesh;
+  if (this->hasConstraint(Mapping::CONSERVATIVE)) {
+    inMesh  = this->output();
+    outMesh = this->input();
+  } else { // Consistent or scaled consistent
+    inMesh  = this->input();
+    outMesh = this->output();
+  }
+
+  // Step 1: get a tentative partitioning consisting of centers and a radius from one of the available algorithms
+  auto [averagePartitionRadius_, centerCandidates] = partitioner::createUniformBlockPartitioning(inMesh, outMesh, _relativeOverlap, _verticesPerPartition, _projectToInput);
+
+  averagePartitionRadius = averagePartitionRadius_;
+  PRECICE_ASSERT(averagePartitionRadius > 0);
+
+  // Step 2: check, which of the resulting partitions are be non-empty (in term sof the output vertices) and register the partition centers in a mesh
+  mesh::Mesh centerMesh("partitionCentersMesh", this->getDimensions(), mesh::Mesh::MESH_ID_UNDEFINED);
```

Comment:
Should we add the name of the input mesh here? What happens if multiple PoU mappings are used?

Reply:
It's only a local variable used for the mapping construction. I don't see any conflicts with multiple instantiations. Am I missing something?

Reply:
We could export the mesh for example (don't you have an option for this already?)
Could also help for debugging in general

Reply:
I have such an option here, yes. Adding the input mesh might be insufficient, though, as the input mesh is 'changes' depending on the constraint.
Exporting the mesh is IMO mostly a developer debugging feature and adding the name is quite easy, so I just added it.

Reply:
Wait: the name for the export mesh is determined in a different place. Right now it is always `pouCenters.vtk`.

Reply:
Changed it now to consider the input mesh name, might conflict for the classical `consistent`/`conservative` case though.